### PR TITLE
Breakup config for pg and srv

### PR DESCRIPTION
**Pull request by nyurik**
Getting ready for multi-backend system, where each backend would have its own configuration.

All 3 config structs (clap args, config, and configbuilder) are now separated into 3 files - the main one, one for service  (actix stuff), and one for PostgreSQL.

The files have been moved to `/pg` and `/srv`, but otherwise the logic should be mostly intact.

### CODE REVIEW

**Comment by maxammann** [comment]
I should have time tomorrow evening to review.

File: src/bin/main.rs [hunk 1]
```diff
@@ -17,30 +18,13 @@ pub struct Args {
...
-              short, long)]
-    pub keep_alive: Option<usize>,
-    #[arg(help = format!("The socket address to bind. [DEFAULT: {}]", ConfigBuilder::LISTEN_ADDRESSES_DEFAULT),
-          short, long)]
-    pub listen_addresses: Option<String>,
-    #[arg(help = format!("Maximum connections pool size [DEFAULT: {}]", ConfigBuilder::POOL_SIZE_DEFAULT),
-          short, long)]
-    pub pool_size: Option<u32>,
-    /// Scan for new sources on sources list requests
+    /// [Deprecated] Scan for new sources on sources list requests
```
Comment (maxammann) [inline src/bin/main.rs:21]: Is the watch flag deprecated? If yes then a warning should be printed.

File: src/config.rs [hunk 2]
```diff
@@ -1,93 +1,44 @@
...
-    pub function_sources: Option<FunctionSources>,
+    #[serde(flatten)]
+    pub srv: srv::config::ConfigBuilder,
+    #[serde(flatten)]
+    pub pg: pg::config::ConfigBuilder,
 }
 
 /// Update empty option in place with a non-empty value from the second option.
-fn set_option<T>(first: &mut Option<T>, second: Option<T>) {
+pub(crate) fn set_option<T>(first: &mut Option<T>, second: Option<T>) {
```
Comment (maxammann) [inline src/config.rs]:
```suggestion
pub fn set_option<T>(first: &mut Option<T>, second: Option<T>) {
```

File: src/pg/mod.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+pub mod composite_source;
```
Comment (maxammann) [inline src/pg/mod.rs:1]: Should `pg` maybe already be called `database`? Or tile source?

**Review by maxammann — APPROVED** [review]

File: src/bin/main.rs (hunk 1 above)
Comment (nyurik) [reply src/bin/main.rs:21]: This flag is hidden from help. There is a depreciation notice if it gets used

File: src/pg/mod.rs (hunk 3 above)
Comment (nyurik) [reply src/pg/mod.rs:1]: I'm working on adding other backends beyond pg

**Pull request merged**